test/validate-syntaxes.js

```diff
@@ -0,0 +1,76 @@
+var properties = require('../css/properties');
+var syntaxes = require('../css/syntaxes');
+var basicTypes = require('../css/types');
+var parseGrammar = require('css-tree').grammar.parse;
+
+var hasErrors = false;
+var itemsHaveErrors;
+
+function isBasicType(name) { return basicTypes.hasOwnProperty(name); }
+function isSyntax(name) { return syntaxes.hasOwnProperty(name); }
+function isProperty(name) { return properties.hasOwnProperty(name); }
+
+function validateGrammar(grammar) {
+  switch (grammar.type) {
+    case 'Group':
+      return grammar.terms
+        .map(validateGrammar)
+        .reduce(function(a, b) { return a.concat(b); }, []);
+    case 'Multiplier':
+      return validateGrammar(grammar.term);
+    case 'Type':
+      var typeName = grammar.name;
+      if (!(isBasicType(typeName) || isSyntax(typeName))) {
```

Comment:
So I guess to accommodate the `inset()` case this will also need to look in `properties`?

Reply:
The `inset()` case was fixed with #246. We only need to look in `properties` when the `grammar.type` is `'Property'`, which correponds to cases like `<'border-radius'>`.